packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/views/explorer/services/DeleteRepresentationTreeItemEventHandler.java

```diff
@@ -34,15 +40,30 @@
 @Service
 public class DeleteRepresentationTreeItemEventHandler implements IDeleteTreeItemHandler {
 
+    private final IRepresentationMetadataDeletionService representationMetadataDeletionService;
+
+    public DeleteRepresentationTreeItemEventHandler(IRepresentationMetadataDeletionService representationMetadataDeletionService) {
+        this.representationMetadataDeletionService = Objects.requireNonNull(representationMetadataDeletionService);
+    }
+
     @Override
     public boolean canHandle(IEditingContext editingContext, TreeItem treeItem) {
         return treeItem.getKind().startsWith(IRepresentation.KIND_PREFIX);
     }
 
     @Override
     public IStatus handle(IEditingContext editingContext, TreeItem treeItem, Tree tree) {
-        Map<String, Object> parameters = new HashMap<>();
-        parameters.put(EditingContextEventProcessor.REPRESENTATION_ID, treeItem.getId());
-        return new Success(ChangeKind.REPRESENTATION_TO_DELETE, parameters);
+        var optionalRepresentationUUID = new UUIDParser().parse(treeItem.getId());
+        var input = new DeleteRepresentationInput(UUID.randomUUID(), treeItem.getId());
```

Comment:
Arf, we should be able to retrieve the proper DeleteTreeItemInput here instead of inventing something. Also you are not disposing the representation here.